web-services/query/src/main/java/datawave/webservice/query/metric/QueryMetricsWriter.java

```diff
@@ -189,158 +206,145 @@ public long getAge() {
         }
     }
     
-    @Schedule(hour = "*", minute = "*", second = "*/10", persistent = false)
-    public void processQueryMetrics() {
-        if (queryMetricsWriterConfiguration.getUseRemoteService()) {
-            useRemoteService();
-        } else {
-            useQueryMetricHandler();
+    private class MetricProcessor implements Runnable {
+        @Override
+        public void run() {
+            // create JMSContext inside the ManagedThread so that the
+            // session is active when we use it to create a consumer
+            JMSContext jmsContext = connectionFactory.createContext(JMSContext.AUTO_ACKNOWLEDGE);
+            boolean useRemoteService = queryMetricsWriterConfiguration.getUseRemoteService();
+            while (!shuttingDown) {
+                try {
+                    // only process metrics if they are being successfully written
+                    if (failedMetrics.isEmpty()) {
+                        List<QueryMetricHolder> metricQueue = getMetricsFromQueue(jmsContext);
+                        if (metricQueue.isEmpty()) {
+                            Thread.sleep(5000);
+                        } else {
+                            if (useRemoteService) {
+                                processQueryMetricsWithRemoteService(metricQueue);
+                            } else {
+                                processQueryMetricsWithHandler(metricQueue);
+                            }
+                        }
+                    } else {
+                        boolean success = writeFailedMetrics();
+                        if (!success) {
+                            // prevent a tight loop where we keep failing to write failedMetrics
+                            Thread.sleep(60000);
+                        }
+                    }
+                } catch (Exception e) {
+                    log.error(e.getMessage(), e);
+                }
+            }
         }
     }
     
-    private void useRemoteService() {
-        if (receivingMetrics.compareAndSet(false, true)) {
+    private void processQueryMetricsWithRemoteService(List<QueryMetricHolder> metricHolderQueue) {
+        List<BaseQueryMetric> metricQueue = metricHolderQueue.stream().map(h -> h.getQueryMetric()).collect(Collectors.toList());
```

Comment:
`h -> h.getQueryMetric()` could be replaced with `QueryMetricHolder::getQueryMetric`